src/createXcmTypes/util/getSystemChainAssetId.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+
+import { ApiPromise } from '@polkadot/api';
+
+import { SYSTEM_PARACHAINS_IDS } from '../../consts';
+import { BaseError } from '../../errors';
+import { Registry } from '../../registry';
+import { foreignAssetMultiLocationIsInRegistry } from './foreignAssetMultiLocationIsInRegistry';
+import { foreignAssetsMultiLocationExists } from './foreignAssetsMultiLocationExists';
+import { getChainIdBySpecName } from './getChainIdBySpecName';
+/**
+ * Returns the correct asset id for a valid system chain token symbol
+ * integer id or foreign asset multilocation
+ * errors if given an invalid symbol, integer id or multilocation that is
+ * not registered or found in the chains state
+ *
+ * @param tokenSymbol string
+ * @param specName string
+ */
+export const getSystemChainAssetId = async (
+	asset: string,
+	specName: string,
+	_api: ApiPromise,
```

Comment:
This one still needs the API to be first.

Reply:
great catch! Updating now :)